src/test/mir-opt/multiple_return_terminators.test.MultipleReturnTerminators.diff

```diff
@@ -29,12 +31,12 @@
                                            // mir::Constant
                                            // + span: $DIR/multiple_return_terminators.rs:4:10: 6:6
                                            // + literal: Const { ty: (), val: Value(Scalar(<ZST>)) }
--         goto -> bb3;                     // scope 0 at $DIR/multiple_return_terminators.rs:4:5: 8:6
--     }
-- 
--     bb3: {
--         return;                          // scope 0 at $DIR/multiple_return_terminators.rs:9:2: 9:2
-+         return;                          // scope 0 at $DIR/multiple_return_terminators.rs:4:5: 8:6
+          goto -> bb3;                     // scope 0 at $DIR/multiple_return_terminators.rs:4:5: 8:6
+      }
+  
+      bb3: {
+          StorageDead(_2);                 // scope 0 at $DIR/multiple_return_terminators.rs:9:1: 9:2
```

Comment:
ugh, ok, this storage dead messes with the optimization. maybe we should run it both in the new location and the old one?

Reply:
This is purely for curiosity, where can I learn more about what these statements represent? I found https://rustc-dev-guide.rust-lang.org/mir/index.html and the associated RFC, is there anything else? :)

Reply:
These are hints for llvm so it can reuse stack space. For example:

```rust
let x = 42;
// do something with x
let y = 55;
// never use x again
```

here we want to be able to use the same stack slot for `y` that we had for `x`. We're considering a more after-the-fact approach that computes this storage information on the fly, but for now that's what we have.

I think there are other reasons, possibly borrowck related, and const evaluation also is happy to use them as an allocation optimization.

Reply:
Thanks for the details! I thought these kind of things were computed automatically by liveness analysis, but I can see how these would be useful in other compiler passes